### PR TITLE
BUG:syncnode for peerinfo and Optimize p2p service flag

### PR DESCRIPTION
**Pull request by lochjin**

### CODE REVIEW

File: qx/handle_base58.go [hunk 1]
```diff
@@ -63,9 +63,7 @@ func Base58CheckEncode(version []byte, mode string, hasher string, cksumSize int
 	}
 	// Show the encoded data.
 	//fmt.Printf("Encoded Data ver[%v] : %s\n",ver, encoded)
-	if err != nil {
-		fmt.Errorf("%s", err)
-	} else {
+	if err == nil {
```
Comment (dindinw) [inline qx/handle_base58.go:66]: Why need to change the qx , how qx affects the p2p?

Comment (lochjin) [reply qx/handle_base58.go:66]: image

Comment (dindinw) [reply qx/handle_base58.go:66]: oh, my bad at https://github.com/Qitmeer/qitmeer/commit/e88f1fc0d953417e2bbdaab3c97da5523e4a7b1b
should change to
```golang
if err != nil {
	ErrExit(err)
}
fmt.Printf("%s\n", encoded)
```

Comment (lochjin) [reply qx/handle_base58.go]: https://github.com/Qitmeer/qitmeer/pull/400/commits/f2b2212bc0161ba6b0c0ab621e934c612c0d6308

**Review by dindinw — APPROVED** [review]

**Pull request merged**